### ISSUE TITLE
BXL: format `ctx.output.print_json()` arguments

### ISSUE THREAD

**Issue by cbarrete** [facebook/buck2#759]
Objects like `cmd_args` or `RunInfo` are formatted when dumped with `ctx.output.write_json()`. Is there a way to do the same thing with `ctx.output.print_json()`? If not, would this be accepted as a new feature, via an optional argument that defaults to `False`?

**Comment by Nero5023** [comment]
You mean [`actions.write_json`](https://buck2.build/docs/api/bxl/actions/#actionswrite_json) here for `ctx.output.write_json()`?

I am not very sure what do you mean here. You mean you want the feature that the output of `ctx.output.print_json()` for objects like `cmd_args` or `RunInfo` be the same as `actions.write_json`?

Right now the output of `ctx.output.print_json()` for object `cmd_args(["echo", "hello"])` is 
```
{
  "items": [
    "\"echo\"",
    "\"hello\""
  ],
  "hidden": [],
  "options": null
}
```

`actions.write_json`: 
```
["echo","hello"]
```

**Comment by cbarrete** [comment]
> You mean [actions.write_json](https://buck2.build/docs/api/bxl/actions/#actionswrite_json) here for ctx.output.write_json()?

Yes I do, my bad.

> I am not very sure what do you mean here. You mean you want the feature that the output of ctx.output.print_json() for objects like cmd_args or RunInfo be the same as actions.write_json?

Yes, I made a POC at https://github.com/facebook/buck2/compare/main...cbarrete:buck2:print-json. I'm sure it wouldn't be merged as is because it just makes a lot of private APIs public for convenience, but the point is that it renders e.g. `cmd_args` as strings or lists of strings, artifacts as paths, etc.

I think that this behavior makes sense for most use cases? If someone is to output data as JSON, surely it is for something else to consume it, and the debug representation of starlark objects is unusable for those.

A concrete example is compilation database generation in C++: without this feature, one needs to write the JSON out to a file, print its path and have the user copy that file, as opposed to "just" redirecting stdout to the desired location.

**Comment by Nero5023** [comment]
Yeah, it makes sense. I will look into this. Thank you for reporting.
